pkg/clusters/addons/kong/kong.go

```diff
@@ -128,28 +148,88 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		return fmt.Errorf("%s: %w", stderr.String(), err)
 	}
 
+	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
 	// configure for dbmode options
 	if a.dbmode == PostgreSQL {
-		a.deployArgs = append(a.deployArgs,
+		dbmodedeployArgs := []string{
 			"--set", "env.database=postgres",
 			"--set", "postgresql.enabled=true",
 			"--set", "postgresql.postgresqlUsername=kong",
 			"--set", "postgresql.postgresqlDatabase=kong",
 			"--set", "postgresql.service.port=5432",
-		)
+		}
+		a.deployArgs = append(a.deployArgs, dbmodedeployArgs...)
 	}
+
 	if a.proxyOnly {
-		a.deployArgs = append(a.deployArgs,
+		proxyOnlyArgs := []string{
 			"--set", "ingressController.enabled=false",
 			"--set", "ingressController.installCRDs=false",
 			"--skip-crds",
-		)
+		}
+		a.deployArgs = append(a.deployArgs, proxyOnlyArgs...)
+		args = append(args, proxyOnlyArgs...)
+	}
+
+	if a.adminServiceTypeLoadBalancer {
+		adminServiceTypeLoadBalancerArgs := []string{"--set", "admin.type=LoadBalancer"}
+		a.deployArgs = append(a.deployArgs, adminServiceTypeLoadBalancerArgs...)
+		args = append(args, adminServiceTypeLoadBalancerArgs...)
+	}
+
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
 	}
 
 	// do the deployment and install the chart
-	args := []string{"--kubeconfig", kubeconfig.Name(), "install", DefaultDeploymentName, "kong/kong"}
-	args = append(args, "--create-namespace", "--namespace", a.namespace)
-	args = append(args, a.deployArgs...)
+
+	args = append(args, "--namespace", a.namespace)
+	if a.enterprise {
+		if a.enterpriseLicenseJSONString == "" {
+			a.enterpriseLicenseJSONString = os.Getenv("KONG_ENTERPRISE_LICENSE")
+			if a.enterpriseLicenseJSONString == "" {
+				return fmt.Errorf("license json should not be empty")
+			}
+		}
+
+		if err := deployKongEnterpriseLicenseSecret(ctx, cluster, a.namespace, KongEnterpriseLicense, a.enterpriseLicenseJSONString); err != nil {
+			return err
+		}
+
+		if a.kongAdminPassword == "" {
+			pwdLen := 10
+			numDigits := 5
+			numSymbol := 0
+			adminPassword, err := password.Generate(pwdLen, numDigits, numSymbol, false, false)
```

Comment:
Do we log this anywhere though? We may want to, to make it easier to perform your own admin API interactions.

As-is (assuming we don't) we're probably still fine, since:
- You don't need to show it for the controller to use the Secret, which is most of the test cases we care about.
- If users want to know the password for manual interaction, they can just provide one.
- If users want to know the generated password, they can just inspect the Secret.

That all in mind, I don't think it's worth trying to refactor this to expose generated passwords via other means for this PR.

Reply:
We generate the password and use it later as kong_password setting. Agree, this PR do not need refactor this part.